### ISSUE TITLE
VM goes to black screen, won't boot back to gui 

### ISSUE THREAD

**Issue by m-c-g** [utmapp/UTM#3859]
**Describe the issue**
Running Debian XFCE, full screen. After the screen blanker engages, VM comes back up with a black screen. After stopping VM and trying to reboot, the reboot is very slow and does not bring up GUI, but allows booting to console.

**Configuration**

* UTM Version: 
3.1.5 (53)
* OS Version: 
Monterey
* Intel or Apple Silicon? 
Macbook Air M1
**Crash log**
None found

**Debug log**
I will include this if I can reproduce the problem

**Pasting the plist**
**Upload VM**

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>ConfigurationVersion</key>
	<integer>2</integer>
	<key>Debug</key>
	<dict>
		<key>DebugLog</key>
		<true/>
	</dict>
	<key>Display</key>
	<dict>
		<key>ConsoleBackgroundColor</key>
		<string>#000000</string>
		<key>ConsoleFont</key>
		<string>Menlo-Regular</string>
		<key>ConsoleFontSize</key>
		<integer>12</integer>
		<key>ConsoleTextColor</key>
		<string>#ffffff</string>
		<key>ConsoleTheme</key>
		<string>Default</string>
		<key>DisplayCard</key>
		<string>virtio-ramfb</string>
		<key>DisplayDownscaler</key>
		<string>linear</string>
		<key>DisplayFitScreen</key>
		<true/>
		<key>DisplayUpscaler</key>
		<string>linear</string>
	</dict>
	<key>Drives</key>
	<array>
		<dict>
			<key>DriveName</key>
			<string>drive0</string>
			<key>ImagePath</key>
			<string>debian.qcow2</string>
			<key>ImageType</key>
			<string>disk</string>
			<key>InterfaceType</key>
			<string>virtio</string>
		</dict>
		<dict>
			<key>DriveName</key>
			<string>drive1</string>
			<key>ImageType</key>
			<string>cd</string>
			<key>InterfaceType</key>
			<string>usb</string>
			<key>Removable</key>
			<true/>
		</dict>
		<dict>
			<key>DriveName</key>
			<string>drive65D97745-8EB3-4EAB-91A2-9F5E2E9C2AA6</string>
			<key>ImagePath</key>
			<string>efi_vars.fd</string>
			<key>ImageType</key>
			<string>none</string>
			<key>InterfaceType</key>
			<string></string>
		</dict>
	</array>
	<key>Info</key>
	<dict>
		<key>Icon</key>
		<string>debian</string>
		<key>IconCustom</key>
		<false/>
		<key>Notes</key>
		<string>Default user/pass: debian/debian
Root user/pass: root/password</string>
	</dict>
	<key>Input</key>
	<dict>
		<key>InputLegacy</key>
		<false/>
	</dict>
	<key>Networking</key>
	<dict>
		<key>NetworkCard</key>
		<string>rtl8139</string>
		<key>NetworkCardMAC</key>
		<string>5A:1F:39:02:1C:01</string>
		<key>NetworkMode</key>
		<string>emulated</string>
	</dict>
	<key>Printing</key>
	<dict/>
	<key>Sharing</key>
	<dict>
		<key>ClipboardSharing</key>
		<true/>
		<key>DirectoryName</key>
		<string></string>
		<key>DirectorySharing</key>
		<true/>
		<key>Usb3Support</key>
		<true/>
		<key>UsbRedirectMax</key>
		<integer>3</integer>
	</dict>
	<key>Sound</key>
	<dict>
		<key>SoundCard</key>
		<string>intel-hda</string>
		<key>SoundEnabled</key>
		<true/>
	</dict>
	<key>System</key>
	<dict>
		<key>Architecture</key>
		<string>aarch64</string>
		<key>BootDevice</key>
		<string></string>
		<key>BootUefi</key>
		<true/>
		<key>CPU</key>
		<string>default</string>
		<key>CPUCount</key>
		<integer>0</integer>
		<key>ForceMulticore</key>
		<false/>
		<key>ForcePS2Controller</key>
		<false/>
		<key>JITCacheSize</key>
		<integer>0</integer>
		<key>MachineProperties</key>
		<string>highmem=off</string>
		<key>Memory</key>
		<integer>1024</integer>
		<key>RTCUseLocalTime</key>
		<true/>
		<key>RngEnabled</key>
		<true/>
		<key>SystemUUID</key>
		<string>8169C503-2646-4317-BE65-7F931B70884F</string>
		<key>Target</key>
		<string>virt</string>
		<key>UseHypervisor</key>
		<true/>
	</dict>
</dict>
</plist>

**Comment by conath** [comment]
When force-closing the VM, the hard disk file may have become corrupted. As far as I know this can be seen by ending up with filesystem errors shown when booting in console mode. Can you see such errors?

**Comment by cpach** [comment]
Is there any reasonable safeguards that could be added to UTM to prevent the corruption of the disk file?

E.g. perhaps UTM could send the ACPI shutdown signal to the guest, so that the guest can shutdown gracefully?

**Comment by conath** [comment]
@cpach I like that idea! Is there a QEMU Monitor command for that?

It seems like #3842 is a similar if not the same issue.

**Comment by conath** [comment]
@cpach indeed, found an example in Canonical Multipass https://github.com/canonical/multipass/blob/9c51de654ea4ff4ca668a3b9852f5e2f89821130/src/platform/backends/qemu/qemu_virtual_machine.cpp

@osy I think we should offer choice of graceful or forced shutdown (using above QMP API) to fix this issue. It remains to be seen if this works for all guest systems.

**Comment by kkai-gg** [comment]
As I understand the disk is the debian.qcow2 file, right?
Is it possible to get the necessary files from it? It is very important!

**Comment by osy** [comment]
Closing this as the ACPI shutdown option was added and the corruption issue unfortunately cannot be fixed.

**Comment by michaelerule** [comment]
Granted, this was closed as unfixable, but this issue was the main result that I found when searching for a solution as of 2025.03.21. 

Is it possible to somehow treat this as a sort of "documentation bug" and add some more information about how one might *recover* lost data from an instance that becomes unbootable due to this issue? 

I'm at a loss as to how to debug this on a Debian 12 image: It seems to reach the bootloader but all available kernels (and their recovery options) boot to a black screen with a blinking cursor. Alt+F(1--12) don't seem to do anything (or rather fn+Alt, since alt keys are defaulted to control actions on the macos host), and I suspect it is because this does not work (it cannot even boot to a working terminal),  not because they keystrokes are not getting through. I need to recover the corrupted image, as close to the most recent viable state as possible.
